src/http/one/TeChunkedParser.cc

```diff
@@ -115,55 +116,61 @@ Http::One::TeChunkedParser::parseChunkSize(Http1::Tokenizer &tok)
  * ICAP 'use-original-body=N' extension is supported.
  */
 bool
-Http::One::TeChunkedParser::parseChunkExtension(Http1::Tokenizer &tok, bool skipKnown)
+Http::One::TeChunkedParser::parseChunkExtensions(::Parser::Tokenizer &tok, bool skipKnown)
+{
+    while (!tok.atEnd()) {
+        if (skipLineTerminator(tok)) {
+            parsingStage_ = theChunkSize ? Http1::HTTP_PARSE_CHUNK : Http1::HTTP_PARSE_MIME;
+            return true; // reached the end of extensions (if any)
+        }
+
+        if (parseChunkExtension(tok, skipKnown)) {
+            buf_ = tok.remaining();
+            continue; // got one extension; there may be more
+        }
+
+        static const CharacterSet NonLF = (CharacterSet::LF).complement().rename("non-LF");
+        if (tok.skipAll(NonLF) && tok.skip('\n'))
+            throw TexcHere("cannot parse chunk extension"); // <garbage> CR*LF
+
+        return false; // need more data to finish parsing extension
+    }
+    return false; // need data to start parsing extension or CRLF
```

Comment:
AFAICT, you used `the` instead of the requested `an` article. IMO, `an` is more appropriate in this context because, unlike the other/earlier `return false` case, we do not know what extension we are going to parse when more data comes in.

Reply:
I used `the` because we are going to parse the extension belonging to the specific chunk (which size we should have parsed already). However, if you insist, I can use `an` here.

Reply:
Yes, I continue to insist. A chunk may have many extensions.

Reply:
Done at 781dbe5.